lessons/templates/lesson_page.html

```diff
@@ -23,32 +23,34 @@
                 <th>Notes</th>
                 <th>Request status</th>
             </tr>
-            {% for user in object_list %}
+            {% for form in object_list %}
             <tr>
-                <td>{{user.student.username}}</td>
-                <td>{{user.availableDays}}</td>
-                <td>{{user.availableTimes}}</td>
-                <td>{{user.numberOfLessons}}</td>
-                <td>{{user.IntervalBetweenLessons}}</td>
-                <td>{{user.DurationOfLesson}}</td>
-                <td>{{user.LearningObjectives}}</td>
-                <td>{{user.AdditionalNotes}}</td>
-                <td>{{user.Fulfilled}}</td>
-                {% if currentUser == True %}
-                {% if user.Fulfilled == 'Pending' %}
-                <td><a href="{% url 'lesson_request_update' user.id %}">Edit</a></td>
-                <td><a href="{% url 'lesson_request_delete' user.id%}">Delete</a></td>
+                <td>{{form.student.username}}</td>
+                <td>{{form.availableDays}}</td>
+                <td>{{form.availableTimes}}</td>
+                <td>{{form.numberOfLessons}}</td>
+                <td>{{form.IntervalBetweenLessons}}</td>
+                <td>{{form.DurationOfLesson}}</td>
+                <td>{{form.LearningObjectives}}</td>
+                <td>{{form.AdditionalNotes}}</td>
+                <td>{{form.Fulfilled}}</td>
+                {% if user.isStudent %}
+                {% if form.Fulfilled == "Pending" %}
+                <td><a href="{% url 'lesson_request_update' form.id %}">Edit</a></td>
+                <td><a href="{% url 'lesson_request_delete' form.id%}">Delete</a></td>
                 {% endif %}
                 {% endif %}
-                {% if currentUser == False %}
-                <td><a href="{% url 'lesson_request_update' user.id %}">Edit</a></td>
-                <td><a href="{% url 'lesson_request_delete' user.id%}">Delete</a></td>
+                {% if user.isStudent == False %}
```

Comment:
Here I think you might have misunderstood what I meant. Even though you made the approve/deny buttons like I suggested, I didn't mean that you should delete the edit/delete buttons for the admin or director.

So they should always be able to edit/delete a request, however, the approve/deny buttons should not appear after the lesson is approved or denied. I think just a simple if statement to check if the lesson is pending should do it.